adapters/celestia/src/verifier/address.rs

```diff
@@ -59,12 +93,24 @@ impl FromStr for CelestiaAddress {
     type Err = CelestiaAddressFromStrError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        // This could be the way to save memory:
-        let (hrp, _raw_address_u5, _variant) = bech32::decode(s)?;
+        let (hrp, raw_address_u5, variant) = bech32::decode(s)?;
         if hrp != HRP {
             return Err(CelestiaAddressFromStrError::InvalidHumanReadablePrefix(hrp));
         }
-        let value = s.as_bytes().to_vec();
+        if variant != VARIANT {
+            return Err(CelestiaAddressFromStrError::InvalidVariant);
+        }
+        if raw_address_u5.len() != 32 {
+            return Err(CelestiaAddressFromStrError::InvalidBech32(
+                bech32::Error::InvalidLength,
+            ));
+        }
```

Comment:
I think this check is incorrect. The final (`u8`) length should be 32, not the intermediate length in (`u5`)

Reply:
Discussed over zoom